### PR TITLE
[NL] Generate page config based on data availability and place relation

### PR DESCRIPTION
**Pull request by shifucun**
* Extend Tile to take optional `places` field so it can render from given places
* Put page config generation logic in func build_page_config(). Later on, this will extends to other query shape.

### CODE REVIEW

**Review by jehangiramjad — APPROVED** [review]
The Python changes look fine to me. Approving to unblock you but I'll let the others look at the tsx code :)

**Review by beets — APPROVED** [review]
very cool!

File: server/config/subject_page.proto
Comment (shifucun) [reply server/config/subject_page.proto]: Done

File: static/js/components/tiles/bar_tile.tsx
Comment (shifucun) [reply static/js/components/tiles/bar_tile.tsx]: Done, and updated to `comparison_places`

File: server/lib/nl_page_config.py
Comment (shifucun) [reply server/lib/nl_page_config.py:67]: Done

**Review by shifucun — COMMENTED** [review]
thanks for review!

**Pull request merged**
